### PR TITLE
Improve doc of element-wise sigmoid logistic function

### PR DESCRIPTION
**Pull request by zori**
This PR improves the documentation of the element-wise sigmoid logistic function by adding a usage example.

Adding plots using sphinx extensions will be left for future PRs :).

EDIT: previously `[WIP] Add example usage to docstrings of functions and links`:
PRs welcome!

Part of improving documentaion effort. See also https://github.com/pfnet/chainer/issues/1867

TODO (finish this list):

Functions:
- [ ] linear
- [x] sigmoid
- [ ] ...

Links:
- [ ] Bilinear
- [ ] Maxout
- [ ] ...

### CODE REVIEW

**Comment by mitmul** [comment]
Very good stating point. Some points I got as feedbacks:

- You can check the output by just putting the estimated output after it instead of using `assert` as follows:
```
>>> y.shape
(3, 4)
```
Writing this line `(3, 4)` enables the doctest to check the output of `y.shape` is the same as that line.

- Let's think about adding plots to show the difference of functions effectively like:
    - [numpy.exp](https://docs.scipy.org/doc/numpy/reference/generated/numpy.exp.html)
    - [Transfer Function Layers (from torch.nn)](https://github.com/torch/nn/blob/master/doc/transfer.md#nn.transfer.dok)

We can add emmbedded plots using sphinx extensions: http://matplotlib.org/sampledoc/extensions.html#inserting-matplotlib-plots

**Comment by zori** [comment]
This PR is ready for review.

**Comment by mitmul** [comment]
LGTM

**Comment by mitmul** [comment]
Let's split PRs for each function to improve docstrings for them.
